packages/snap/src/utils/prompt.ts

```diff
@@ -1,20 +1,89 @@
-import { addressPage, balancePage } from '../dialogs';
-import { RpcMethods } from '@hathor/hathor-rpc-handler';
+import { addressPage, balancePage, changeNetworkPage, createNanoPage, createTokenPage, oracleDataPage, sendTransactionPage, signWithAddressPage, utxosPage } from '../dialogs';
+import { setNetwork } from '../utils/network';
+import { NETWORK_MAP } from '../constants';
+import { RpcMethods, TriggerTypes } from '@hathor/hathor-rpc-handler';
 
-export const promptHandler = (origin) => async (promptRequest) => {
+export const promptHandler = (origin, wallet) => async (promptRequest) => {
   const data = promptRequest.data;
   const params = promptRequest.params;
-  let confirmed;
-  switch (promptRequest.method) {
-    case RpcMethods.GetAddress:
-      confirmed = await addressPage(data, params, origin);
-      break;
-    case RpcMethods.GetBalance:
-      confirmed = await balancePage(data, params, origin);
+  let approved;
+  switch (promptRequest.type) {
+    case TriggerTypes.SignMessageWithAddressConfirmationPrompt:
+      approved = await signWithAddressPage(data, params, origin);
+      return { data: approved };
+    case TriggerTypes.PinConfirmationPrompt:
+      return {
+        data: {
+          accepted: true,
+          pinCode: '123',
+        }
+      };
+    case TriggerTypes.GetBalanceConfirmationPrompt:
+      approved = await balancePage(data, params, origin);
+      return { data: approved };
+    case TriggerTypes.AddressRequestPrompt:
+      approved = await addressPage(data, params, origin);
+      return { data: approved };
+    case TriggerTypes.GetUtxosConfirmationPrompt:
+      approved = await utxosPage(data, params, origin);
+      return { data: approved };
+    case TriggerTypes.ChangeNetworkConfirmationPrompt:
+      if (!(params.newNetwork in NETWORK_MAP)) {
+        // Reject if the newNetwork is not a valid option
+        return { data: false };
+      }
```

Comment:
- Should I move this logic to another file? Maybe each rpc method should have a prompt handler file/method

I can change this in another PR if we decide it's better

Reply:
It doesn't seem necessary to me: the logic is still simple enough to stay in the same file.